File: PYTHON/PROGRAM/Layar.py
# KELAS LAYAR
from Komponen import Komponen

class Layar(Komponen):
    def __init__(self, merk="", nama="", produsen="", ukuran=0, resolusi="", jenis_panel="", touchscreen=""):
        Komponen.__init__(self, merk, nama, produsen)
        self.__ukuran = ukuran
        self.__resolusi = resolusi
        self.__jenis_panel = jenis_panel
        self.__touchscreen = touchscreen
    
    def set_ukuran(self, ukuran):
        self.__ukuran = ukuran
    
    def set_resolusi(self, resolusi):
        self.__resolusi = resolusi
    
    def set_jenis_panel(self, jenis_panel):
        self.__jenis_panel = jenis_panel
    
    def set_touchscreen(self, touchscreen):
        self.__touchscreen = touchscreen
    
    def get_ukuran(self):
        return self.__ukuran
    
    def get_resolusi(self):
        return self.__resolusi
    
    def get_jenis_panel(self):
        return self.__jenis_panel
    
    def get_touchscreen(self):
        return self.__touchscreen
    
    def tampilkan_info(self):
        Komponen.tampilkan_info(self)
        print("Ukuran:", self.__ukuran, "inch")
        print("Resolusi:", self.__resolusi)
        print("Jenis Panel:", self.__jenis_panel)
        print("Touchscreen:", self.__touchscreen)

File: CPP/PROGRAM/Layar.cpp
// KELAS LAYAR
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class Layar : public Komponen {
private:
    double ukuran; // inch
    string resolusi;
    string jenisPanel; // LCD, LED, OLED, dll
    string touchscreen; // Ya/Tidak
public:
    Layar() {}
    
    Layar(string merk, string nama, string produsen, double ukuran, string resolusi, string jenisPanel, string touchscreen) 
        : Komponen(merk, nama, produsen) {
        this->ukuran = ukuran;
        this->resolusi = resolusi;
        this->jenisPanel = jenisPanel;
        this->touchscreen = touchscreen;
    }
    
    void setUkuran(double ukuran) {
        this->ukuran = ukuran;
    }
    
    void setResolusi(string resolusi) {
        this->resolusi = resolusi;
    }
    
    void setJenisPanel(string jenisPanel) {
        this->jenisPanel = jenisPanel;
    }
    
    void setTouchscreen(string touchscreen) {
        this->touchscreen = touchscreen;
    }
    
    double getUkuran() {
        return ukuran;
    }
    
    string getResolusi() {
        return resolusi;
    }
    
    string getJenisPanel() {
        return jenisPanel;
    }
    
    string getTouchscreen() {
        return touchscreen;
    }
    
    void tampilkanInfo() {
        Komponen::tampilkanInfo();
        cout << "Ukuran: " << ukuran << " inch" << endl;
        cout << "Resolusi: " << resolusi << endl;
        cout << "Jenis Panel: " << jenisPanel << endl;
        cout << "Touchscreen: " << touchscreen << endl;
    }
    
    ~Layar() {}
};